pwndbg/heap/ptmalloc.py

```diff
@@ -177,21 +179,61 @@ def bk_nextsize(self):
 
         return self._bk_nextsize
 
+    @property
+    def arena(self):
+        if self._arena is None:
+            try:
+                ar_ptr = pwndbg.heap.current.get_heap(self.address)["ar_ptr"]
+                ar_ptr.fetch_lazy()
+            except Exception:
+                ar_ptr = None
+            if ar_ptr is not None and ar_ptr in (ar.address for ar in pwndbg.heap.current.arenas):
+                self._arena = Arena(ar_ptr)
+            else:
+                self._arena = Arena(pwndbg.heap.current.main_arena.address)
+
+        return self._arena
+
+    @property
+    def is_top_chunk(self):
+        if self._is_top_chunk is None:
+            ar = self.arena
+            if ar is not None and self.address == ar.top:
+                self._is_top_chunk = True
+            else:
+                self._is_top_chunk = False
+
+        return self._is_top_chunk
+
     # TODO Other useful methods e.g. next_chunk(), __iter__, __str__
 
 
 class Arena:
-    def __init__(self, addr, heaps):
-        self.addr = addr
+    def __init__(self, addr, heaps=None):
```

Comment:
```suggestion
    __slots__ = ("_gdbValue", "address", "is_main_arena", "_top", "heaps")

    def __init__(self, addr, heaps=None):
```

Reply:
`__slots__` is an optimization, as per https://docs.python.org/3/reference/datamodel.html#slots:
> __slots__ allow us to explicitly declare data members (like properties) and deny the creation of [__dict__](https://docs.python.org/3/library/stdtypes.html#object.__dict__) and __weakref__ (unless explicitly declared in __slots__ or available in a parent.)
> 
> The space saved over using [__dict__](https://docs.python.org/3/library/stdtypes.html#object.__dict__) can be significant. Attribute lookup speed can be significantly improved as well.

With slots we won't be able to add more fields in runtime to an instance of Arena, but I don't think we need it.

Reply:
Yeah I can apply `__slots__` to the `Chunk` class too if you want?